third_party/xla/ordered_set.h

```diff
@@ -56,12 +56,16 @@ class OrderedSet {
   // set.
   void Erase(T value) {
     auto it = value_to_index_.find(value);
-    assert(it != value_to_index_.end());
+    if (it == value_to_index_.end()) {
+      std::cerr << "Value not found in OrderedSet" << std::endl;
+      exit(0);
+    }
 
+    auto index = it->second;
```

Comment:
it should not be null, as the result of iterator should return .end() anyway, so this may not be necessary; Original code seems to be fine

Reply:
yes but the original work doesn't work with Coverity...

Reply:
`it` comes from the `map` `value_to_index_` and to assign its value to its own container might be "self-modification", therefore I placed the value in `index` and then assign it.